### PR TITLE
fix: Allow multiple line in the MessageInput of Mobile env

### PR DESCRIPTION
**Pull request by HoonBaek**
[CLNP-2319](https://sendbird.atlassian.net/browse/CLNP-2319)

### Issue
In a desktop layout, prevent line breaks from occurring even when the 'Enter' key is pressed.
It triggers sending the message.

But in the Mobile layout, this behavior doesn't fit mobile UX.
So need to allow line breaks when users press the 'Enter' key.

### Fix & ChangeLog
* Add a prop `isMobile` to the ui/MessageInput
* Fix the `updateMessage` logic in the MessageInput to apply the line break

[CLNP-2319]: https://sendbird.atlassian.net/browse/CLNP-2319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bang9** [comment]
It would be a good idea to check cross-browser compatibility!

**Comment by HoonBaek** [comment]
@bang9 Thanks for the good suggestion! I confirmed it's resolved in some mobile browsers
tested on: iPhone14pro with Chrome, Firefox, Edge, Opera, and Safari

File: src/ui/MessageInput/index.tsx [hunk 1]
```diff
@@ -410,8 +412,13 @@ const MessageInput = React.forwardRef<HTMLInputElement, MessageInputProps>((prop
           const { userid = '' } = dataset;
           messageText += innerText;
           mentionTemplate += `${USER_MENTION_TEMP_CHAR}{${userid}}`;
+        } else if (node.nodeType === NodeTypes.ElementNode && node.nodeName === NodeNames.Br) {
           messageText += '\n';
           mentionTemplate += '\n';
+        } else if (node?.nodeType === NodeTypes.ElementNode && node?.nodeName === NodeNames.Div) {
+          const { textContent = '' } = node;
+          messageText += `\n${textContent}`;
+          mentionTemplate += `\n${textContent}`;
```
Comment (bang9) [inline src/ui/MessageInput/index.tsx:421]: 혹시 sendMessage 랑 editMessage 차이점이 뭔가요?

Comment (bang9) [reply src/ui/MessageInput/index.tsx:421]: 아 textField.childNodes.forEach 로직이 같은건가 궁금해서요

Comment (liamcho) [reply src/ui/MessageInput/index.tsx:421]: Screenshot 2024-02-21 at 3 44 52 PM
요거만 이슈가 있나요? 메세지 인풋쪽에는 픽스가 필요없나요?

Comment (liamcho) [reply src/ui/MessageInput/index.tsx:421]: 아 아래 픽스가 적용됐군요!
`&& !isMobile`

**Review by liamcho — APPROVED** [review]
LGTM!

File: src/ui/MessageInput/utils.ts [hunk 2]
```diff
@@ -21,3 +23,33 @@ export const nodeListToArray = (childNodes?: Node['childNodes'] | null) => {
 export function isChannelTypeSupportsMultipleFilesMessage(channel: BaseChannel) {
   return channel && channel.isGroupChannel?.() && !channel.isBroadcast && !channel.isSuper;
 }
+
+// Type guard: This function ensures that the node contains `innerText` and `dataset` properties
+function isHTMLElement(node: ChildNode): node is HTMLElement {
+  return node.nodeType === NodeTypes.ElementNode;
+}
```
Comment (bang9) [inline src/ui/MessageInput/utils.ts:30]: 👍👍

**Review by bang9 — APPROVED** [review]
lgtm! please check lint and build error before merge

**Pull request merged**